src/store/wallet.ts

```diff
@@ -353,6 +358,9 @@ export const actions = actionTree(
       const balances = (balancesResults.filter((token) => token && token.rawBalance.gt(0)) as ZkInBalance[]).sort(utils.compareTokensById);
       const balancesEmpty = (balancesResults.filter((token) => token && token.rawBalance.lte(0)) as ZkInBalance[]).sort(utils.sortBalancesAZ);
       balances.push(...balancesEmpty);
+      if (savedAddress !== this.app.$accessor.account.address) {
+        return localList.list;
+      }
       commit("setTokensList", {
         lastUpdated: new Date().getTime(),
```

Comment:
Move lastUpdate into the setTokensList.
And in general try to use single-access points to get data, e.g. tokens store should be called by 2-4 methods only like:
— getAllTokens()
— getAllowedList()
— isRestricted()

### Why?

1. it's best way to avoid unrequited API-calls: each method should be used as **lazy** e.g. first time called it will require data from the API as a result of empty local property.
2. you have endless opportunities to incapsulate all caching / validating / refreshing approaches and change them without modifications of the other codeparts
3. Splitting the app into solid strict-typed and incapsulated will allow us to:
    — create separate modules optional to require and minify overall amount of duplicated code
    — simplify testing — as this solid separate module can both use real API to get the data or trigger some vectors and the dapp will be ok with that — so we can use know data to assure e2e tests
    — it will simplify each separate solid component / module / plugin of the dapp and improve general stability of every our dapp
4. It's aligned with the conception of UX-proof interface and Dapps checklists like our checklist: https://www.notion.so/matterlabs/A-dApp-Usability-Checklist-dfbe0a65dfa44bdb883836b27899b07c

Reply:
@JackHamer09 create a task for yourself in linear to figure it out